src/test/erasure-code/ceph_erasure_code_non_regression.cc

```diff
@@ -94,10 +95,9 @@ int ErasureCodeNonRegression::setup(int argc, char** argv) {
     ceph_options.push_back(i->c_str());
   }
 
-  global_init(
-    &def_args, ceph_options, CEPH_ENTITY_TYPE_CLIENT,
-    CODE_ENVIRONMENT_UTILITY,
-    CINIT_FLAG_NO_DEFAULT_CONFIG_FILE);
+  cct = global_init(&def_args, ceph_options, CEPH_ENTITY_TYPE_CLIENT,
+		    CODE_ENVIRONMENT_UTILITY,
+		    CINIT_FLAG_NO_DEFAULT_CONFIG_FILE);
```

Comment:
@smithfarm because the `cct` here should be held be by the `ErasureCodeNonRegression` class, in other words, the `ErasureCodeNonRegression` should outlive the `cct` by holding a strong reference of it.